data-prepper-plugins/obfuscate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/obfuscation/ObfuscationProcessor.java

```diff
@@ -94,14 +107,24 @@ public Collection<Record<Event>> doExecute(Collection<Record<Event>> records) {
         for (final Record<Event> record : records) {
             final Event recordEvent = record.getData();
 
+            if (Objects.nonNull(obfuscationProcessorConfig.getObfuscateWhen()) && !expressionEvaluator.evaluateConditional(obfuscationProcessorConfig.getObfuscateWhen(), recordEvent)) {
```

Comment:
Why call `Objects.nonNull` here? This adds an unnecessary function call that could be avoided with `obfuscationProcessorConfig.getObfuscateWhen() != null`. 

Note that the API states this method was added to help with `Predicate`s. https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#nonNull-java.lang.Object-

Reply:
I use it because it looks better, but thanks for explaining I will just check != null